library/src/main/java/com/mapzen/valhalla/Router.kt

```diff
@@ -56,6 +56,7 @@ interface Router {
             city: String? = null,
             state: String? = null): Router
     fun setDistanceUnits(units: DistanceUnits): Router
+    fun setMaxDifficulty(difficulty: Int): Router
```

Comment:
I know the other methods don't have documentation but can we add some for this method so people know which values are valid? Lets take what's listed on the [Valhalla docs](https://valhalla.readthedocs.io/en/latest/api/turn-by-turn/api-reference/)
```
This value indicates the maximum difficulty of hiking trails that is allowed. Values between 0 and 6 are allowed. The values correspond to sac_scale values within OpenStreetMap, see reference here. The default value is 1 which means that well cleared trails that are mostly flat or slightly sloped are allowed. Higher difficulty trails can be allowed by specifying a higher value for max_hiking_difficulty.
```